lib/do_volume/do_delete.js

```diff
@@ -0,0 +1,200 @@
+                    if (!getVolErr) {
+                        ctx.volume = volume;
+                    }
+
+                    next(getVolErr);
+                });
+        },
+        function doDeleteVolume(ctx, next) {
+            assert.object(ctx.volume, 'ctx.volume');
+
+            tritonapi.cloudapi.deleteVolume(ctx.volume.id,
+                next);
+        },
+        function waitForVolumeStates(ctx, next) {
+            assert.object(ctx.volume, 'ctx.volume');
+
+            var distraction;
+            var volumeId = ctx.volume.id;
+
+            if (!opts.wait) {
+                next();
+                return;
+            }
+
+            distraction = distractions.createDistraction(opts.wait.length);
+
+            tritonapi.cloudapi.waitForVolumeStates({
+                id: volumeId,
+                states: ['deleted', 'failed']
+            }, function onWaitDone(waitErr, volume) {
+                distraction.destroy();
+                next(waitErr);
+            });
+        }
+    ], arg: {}}, cb);
+}
+
+function do_delete(subcmd, opts, args, cb) {
+    var self = this;
+
+    if (opts.help) {
+        self.do_help('help', {}, [subcmd], cb);
+        return;
+    } else if (args.length < 1) {
+        cb(new errors.UsageError('missing VOLUME arg(s)'));
+        return;
+    }
+
+    var context = {
+        volumeIds: args,
+        cli: this.top
+    };
+
+    vasync.pipeline({arg: context, funcs: [
+        common.cliSetupTritonApi,
+        function confirm(ctx, next) {
+            var promptMsg;
+
+            if (opts.yes) {
+                next();
+                return;
+            }
+
+            if (ctx.volumeIds.length === 1) {
+                promptMsg = format('Delete volume %s? [y/n] ',
+                    ctx.volumeIds[0]);
+            } else {
+                promptMsg = format('Delete %d volumes? [y/n] ',
+                    ctx.volumeIds.length);
+            }
+
+            common.promptYesNo({msg: promptMsg},
+                function onPromptAnswered(answer) {
+                    if (answer !== 'y') {
+                        console.error('Aborting');
+                        /*
+                         * Early abort signal.
+                         */
+                        next(true);
+                    } else {
+                        next();
+                    }
+                });
+        },
+        function deleteVolumes(ctx, next) {
+            vasync.forEachParallel({
+                func: function doDeleteVolume(volumeId, done) {
+                    self.top.tritonapi.deleteVolume({
+                        id: volumeId,
+                        wait: opts.wait && opts.wait.length > 0,
+                        waitTimeout: opts.wait_timeout * 1000
+                    }, function onVolDeleted(volDelErr) {
+                        if (!volDelErr) {
+                            console.log('Deleted volume %s', volumeId);
```

Comment:
If we didn't `-w` wait, then we haven't yet deleted the volume.
Would be nice to have a "Deleting ..." before the API call if we are
not waiting, and if we *are* waiting, then a "Deleted ..." after returning
from the API call.